src/lua/utils.c

```diff
@@ -735,6 +735,30 @@ luaT_newthread(struct lua_State *L)
 	return L1;
 }
 
+/**
+ * A safer os.getenv() variant. The difference is that it returns the value only
+ * if it has sane length (see src/lib/core/utils.c for details).
+ */
+static int
+luaT_getenv(struct lua_State *L)
+{
+	if (lua_gettop(L) != 1 || !lua_isstring(L, 1))
+		return luaL_error(L, "usage: os.getenv(name)");
```

Comment:
`os.getenv()` accepts second, third and so on arguments (and ignores them). We have no strict policy about unknown positional parameters (ignore for compatibility with future implementations or forbid to prevent mistakes). However here we reflect a behavior of a standard function, so I would reflect it precisely in this aspect.

The standard `os.getenv()` also accepts a `number` (similarly to `lua_tolstring()` and also implicitly changes the number to the string). Personally I consider this behavior disgusting, but it is how Lua works. Maybe we should reflect it as well.

Reply:
Okay:
```diff
diff --git a/src/lua/utils.c b/src/lua/utils.c
index 193072467..870bc4286 100644
--- a/src/lua/utils.c
+++ b/src/lua/utils.c
@@ -742,9 +742,9 @@ luaT_newthread(struct lua_State *L)
 static int
 luaT_getenv(struct lua_State *L)
 {
-       if (lua_gettop(L) != 1 || !lua_isstring(L, 1))
-               return luaL_error(L, "usage: os.getenv(name)");
        const char *name = lua_tostring(L, 1);
+       if (name == NULL)
+               return luaL_error(L, "usage: os.getenv(name)");
        char *envvar = getenv_safe(name, NULL, 0);
        if (envvar != NULL) {
                lua_pushstring(L, envvar);
```